src/core/math.d

```diff
@@ -164,3 +164,76 @@ unittest
     }
 }
 
+/*************************************
+ * Round argument to a specific precision.
+ *
+ * D language types specify a minimum precision, not a maximum. The
+ * `toPrec()` function forces rounding of the argument `f` to the
+ * precision of the specified floating point type `T`.
+ *
+ * Params:
+ *      T = precision type to round to
+ *      f = value to convert
+ * Returns:
+ *      f in precision of type `T`
+ */
+@safe pure nothrow
+T toPrec(T:float)(float f) { pragma(inline, false); return f; }
```

Comment:
Why ` pragma(inline, false);`?

Reply:
I would assume just in case the inliner removes the call before builtins handler is given a chance to expand it.

Reply:
I'm worried about performance, but I suppose if this is a CTFE only intrinsic then there is no problem.